### PR TITLE
fixes #3979 - wrong logic in the TestCase logger setup causes duplicate messages

### LINKED ISSUE #3979: [bug] wrong logic in the TestCase logger setup causes duplicate messages
The current logic in the [_set_worker_logger](https://github.com/SatelliteQE/robottelo/blob/master/robottelo/test.py#L106) fixture causes additional handlers being attached to the logger per each call of `setUpClass`, resulting in huge (hundreds of Megs to Gigabytes) robotello*.log files per tier per build!

- __root cause__: new handler is being created (with new hash) and attached to the logger without checking whether the logger already has such handler attached.

- __proposed fix__: use `handler.set_name()` in order to be able to identify the handler and wrap the attaching logic into a conditional (if such handler already attached => ignore)

- [x] master - https://github.com/SatelliteQE/robottelo/pull/3980
- [x] 6.2.z - https://github.com/SatelliteQE/robottelo/pull/3981

### PR DESCRIPTION
**Pull request by rplevka**
addresses: https://github.com/SatelliteQE/robottelo/issues/3979

requires cherry-pick to `6.2.z`

### CODE REVIEW

File: robottelo/test.py [hunk 1]
```diff
@@ -110,17 +110,20 @@ def _set_worker_logger(self, worker_id):
...
-                datefmt='%Y-%m-%d %H:%M:%S'
-            )
-            handler = logging.FileHandler(
-                'robottelo_{0}.log'.format(worker_id))
-            handler.setFormatter(formatter)
-            self.logger.addHandler(handler)
-            # Nailgun HTTP logs should also be included in gw* logs
-            logging.getLogger('nailgun').addHandler(handler)
+            if not [h for h in self.logger.handlers
+                    if h.get_name() == '{0}'.format(worker_id)]:
```
Comment (rochacbruno) [inline robottelo/test.py]: I guess if it was not more clear if:

```python
if not '{0}'.format(worker_id) in [h.get_name() for h in self.logger.handlers]
```

Comment (rplevka) [inline robottelo/test.py]: :+1:

**Comment by rplevka** [comment]
comment addressed

File: robottelo/test.py [hunk 2]
```diff
@@ -110,17 +110,20 @@ def _set_worker_logger(self, worker_id):
...
-                datefmt='%Y-%m-%d %H:%M:%S'
-            )
-            handler = logging.FileHandler(
-                'robottelo_{0}.log'.format(worker_id))
-            handler.setFormatter(formatter)
-            self.logger.addHandler(handler)
-            # Nailgun HTTP logs should also be included in gw* logs
-            logging.getLogger('nailgun').addHandler(handler)
+            if '{0}'.format(worker_id) not in [
+                    h.get_name()for h in self.logger.handlers]:
```
Comment (rochacbruno) [inline robottelo/test.py]: missing space here `h.get_name()for h in self.logger.handlers`

**Comment by rochacbruno** [comment]
ACK

**Comment by omaciel** [comment]
Thanks @rplevka !!!

**Pull request merged**